src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/exception/DownstreamApiCallError.kt

```diff
@@ -0,0 +1,75 @@
+4) Cannot find NSI
+res.body={"status":400,"developerMessage":"Cannot find NSI for CRN: M190420 Sentence: 1503032340 and ContractType ACC"}
+5) Schedule appointment conflict
+res.body=409 Conflict from POST https://community-api-secure.probation.service.justice.gov.uk/secure/offenders/crn/E188275/sentence/1503090023/appointments/context/commissioned-rehabilitation-services
+6) Reschedule appointment conflict
+res.body=409 Conflict from POST https://community-api-secure.probation.service.justice.gov.uk/secure/offenders/crn/D889766/appointments/1761440075/reschedule/context/commissioned-rehabilitation-services
+
+Error categories after (Replace [A-Z]*[0-9]+ with _ and Replace '[A-Z0-9]*' with '_')
+=====================================================================================
+1) Contact type '_' requires an outcome type as the contact date is in the past '_-_-_'
+2) Validation failure: startTime endTime must be after or equal to startTime, endTime endTime must be after or equal to startTime"
+3) CRN: _ EventId: _ has multiple referral requirements
+4) Cannot find NSI for CRN: _ Sentence: _ and ContractType ACC
+5) _ Conflict from POST https://community-api-secure.probation.service.justice.gov.uk/secure/offenders/crn/_/sentence/_/appointments/context/commissioned-rehabilitation-services
+6) _ Conflict from POST https://community-api-secure.probation.service.justice.gov.uk/secure/offenders/crn/_/appointments/_/reschedule/context/commissioned-rehabilitation-services*/
+
+class DownstreamApiCallError(val httpStatus: HttpStatus, causeMessage: String, val responseBody: String, val exception: Throwable) :
+  RuntimeException(
+    exception
+  ) {
+  val category = errorCategoryByRemovingIdentifiers(causeMessage)
+  private val error = when {
+    httpStatus.is4xxClientError -> {
+      when {
+        category.contains("Contact type .* requires an outcome type as the contact date is in the past".toRegex()) -> {
+          Pair("An appointment must only be made for today or in the future. Please try again", false)
+        }
+        category.contains("endTime must be after or equal to startTime".toRegex()) -> {
+          Pair("An appointment must complete within the same day. Please try again", false)
+        }
+        category.contains("CRN: _ EventId: _ has multiple referral requirements".toRegex()) -> {
+          Pair("The Service User Sentence must not contain more than one active rehabilitation activity requirement. Please correct and try again", true)
+        }
+        category.contains("Cannot find NSI for CRN: _ Sentence: _ and ContractType".toRegex()) -> {
+          Pair("The intervention cannot be found for the Service User Sentence. Please contact support", true)
+        }
+        category.contains("Multiple existing matching NSIs found".toRegex()) -> {
+          Pair("Multiple NSI's match this referral. Please contact support", true)
```

Comment:
To encourage the reduction in support queries (and because we'd action this):
```suggestion
          Pair("There has been an error during creating the Delius NSI. We are aware of this issue. For follow-up, please contact support", true)
```

Reply:
Done